util/src/main/java/com/google/cloud/hadoop/util/GcsJsonApiEvent.java

```diff
@@ -116,4 +116,67 @@ private GcsJsonApiEvent set(String key, Object value) {
     this.properties.put(key, value);
     return this;
   }
+
+  private static RequestType getRequestType(HttpResponse httpResponse) {
+    HttpRequest request = httpResponse.getRequest();
+    String method = request.getRequestMethod();
+    if (HttpMethods.GET.equals(method)) {
+      GenericUrl url = request.getUrl();
+
+      // This not a clean way to classify the type of GET operation. Going with this approach since
+      // it is simpler and get the job done.
+      Object maxResult = url.getFirst("maxResults");
+      if (maxResult != null) {
+        String maxResultString = maxResult.toString();
+        if ("1".equals(maxResultString)) {
+          return RequestType.LIST_FILE;
+        }
+
+        return RequestType.LIST_DIR;
+      }
+
+      Object fields = url.getFirst("fields");
+      if (fields != null && fields.toString().contains("metadata")) {
+        return RequestType.GET_METADATA;
+      }
+
+      Object alt = url.getFirst("alt");
+      if (alt != null && "media".equals(alt.toString())) {
+        return RequestType.GET_MEDIA;
+      }
+
+      return RequestType.GET_OTHER;
+    } else if (HttpMethods.DELETE.equals(method)) {
+      return RequestType.DELETE;
+    } else if (HttpMethods.PUT.equals(method)) {
+      return RequestType.PUT;
+    } else if (HttpMethods.POST.equals(method)) {
+      return RequestType.POST;
+    } else if (HttpMethods.PATCH.equals(method)) {
+      return RequestType.PATCH;
+    }
+
+    return RequestType.OTHER;
+  }
+
+  public enum EventType {
+    BACKOFF,
+    EXCEPTION,
+    RESPONSE,
+    RETRYSKIPPED,
+    STARTED,
+  }
+
+  public enum RequestType {
```

Comment:
The reason for this enum is because we would like to have bifurcation of the "GET" methods (i.e. metadata, media etc). We currently are not having bifurcation for PUT. Other thank resumable upload, I am not aware of any PUT? If we want bifurcation we can refactor and add/update the enum.

Reply:
Comment is there in the Enum description.